NitroxServer/Program.cs

```diff
@@ -31,6 +30,30 @@ static void Main(string[] args)
                 return;
             }
 
+            // Catch Exit Event
+            PlatformID platid = Environment.OSVersion.Platform;
+            switch(platid)
```

Comment:
Isn't doing `Console.CancelKeyPress += (sender, args) => { Server.Instance.Stop(); Environment.Exit(0); }` enough for all platforms? Seems to work fine without the additional Win32 DLLImport and code.

Reply:
CancelKeyPress only handle Ctrl+C request, it will not be invoked when User **kill** it (signal 9, or closing console window). You can try closing window on windows to see if cancelkeypress delegate invoked.

SetConsleCtrlHandler will catch that closing.

I tried signal handling in the beginning but its just hard for me to implement on C#.